### PR TITLE
Interactive login

### PR DESCRIPTION
**Pull request by miclaraia**
Using current functionality, logging in requires a user to manually
enter their credentials within their script (using the login() function or
by passing credentials to the Panoptes object initializer), or to store them as
environment variables in `PANOPTES_USERNAME` and `PANOPTES_PASSWORD`.

This commit adds functionality for an interactive login. This would allow a
user to enter their credentials manually when running a script that doesn't
involve storing the credentials in a static file or in an environment
variable.

### CODE REVIEW

**Comment by astopy** [comment]
Thanks for the pull request! This fits pretty well into my plans for version 1.1, so your timing is perfect :)

Storing/using credentials is going to be a common problem for people using this module and it's also come up in the CLI, which uses this module (see also zooniverse/panoptes-cli#56 for some previous discussion of this). My plan was to build something like this into the CLI itself, but now I'm thinking we should go further and add keyring support into the Client and just wrap that in the CLI.

Implementing interactive login first makes sense, and this PR is pretty much all we'd need to do there I think. The only thing I'd add is to maybe make a modification to `Panoptes.connect()` to support this there with something like `Panoptes.connect(login='interactive')`. This would allow future support for manually selecting a specific login method, so we could allow e.g. `keyring`, `interactive`, or `auto`, where `auto` would check the keyring for a password and prompt for one to save if there isn't already one in there.

Would you have time to add the `login='interactive'` option to this PR? `Panoptes.connect` would just need to check the value of `login` and call `interactive_login` if it needs to. No worries if you don't have time, I can merge this and add the rest later.

File: panoptes_client/panoptes.py [hunk 1]
```diff
@@ -101,8 +103,10 @@ def __init__(
             'PANOPTES_ENDPOINT',
             'https://www.zooniverse.org'
         )
-        self.username = username or os.environ.get('PANOPTES_USERNAME')
-        self.password = password or os.environ.get('PANOPTES_PASSWORD')
+        self.username = None
+        self.password = None
+        self._auth(username, password, login)
```
Comment (astopy) [inline panoptes_client/panoptes.py]: Looks like the arguments are in the wrong order here compared with the definition of `_auth`.

**Comment by miclaraia** [comment]
Oops sorry about, should've checked that a little more closely

**Comment by astopy** [comment]
@miclaraia thanks!

**Pull request merged**